src/audio/base_fw.c

```diff
@@ -122,6 +124,16 @@ static int basefw_config(uint32_t *data_offset, char *data)
 
 	tuple = tlv_next(tuple);
 
+
+	priv_caps.privacy_version = 1;
+	priv_caps.capabilities_length = 1;
+	priv_caps.capabilities[0] = mic_privacy_get_policy_register();
+
+	tlv_value_set(tuple, IPC4_PRIVACY_CAPS_HW_CFG, sizeof(priv_caps), &priv_caps);
```

Comment:
This is setting the privacy to the wrong TLV, this is the FW_CONFIG, not the HW_CONFIG.
IPC4_PRIVACY_CAPS_HW_CFG == IPC4_MAX_ASTATE_COUNT_FW_CFG

Reply:
This still sets the mic privacy into the _wrong_ config..

Reply:
@ujfalusi this looks correct now.

Reply:
No, this has to be within basefw_hw_config() to be in correct place.

Reply:
with this change you add _two_ IPC4_MAX_ASTATE_COUNT_FW_CFG to the fw_config data.
Or _two_ IPC4_PRIVACY_CAPS_HW_CFG.

In any case this is incorrect and will cause issues for the host.